x-pack/plugins/index_management/public/application/services/routing.ts

```diff
@@ -78,3 +82,19 @@ export const getComponentTemplatesLink = (usedByTemplateName?: string) => {
   }
   return url;
 };
+export const navigateToIndexDetailsPage = (
+  indexName: string,
+  indicesListURLParams: string,
+  extensionsService: ExtensionsService,
+  application: ApplicationStart,
+  http: HttpSetup,
+  history: ScopedHistory,
+  tabId?: IndexDetailsSection
+) => {
+  if (!extensionsService.indexDetailsPageRoute) {
+    history.push(getIndexDetailsLink(indexName, indicesListURLParams, tabId));
```

Comment:
This doesn't necessarily need to be addressed in this PR but its a bit odd that we use `history.push` in one place and `application.navigateToUrl` in another.

Reply:
yeah, `search_indices` is registered as a plugin and uses indexManagement `extensionService.indexDetailsPageRoute` to register search_indices index details route.  I used `application.navigateToUrl`  for navigating to this route  as search_indices is a different app whereas history.push does not help in this scenario. 

I will try finding out other approaches. But meanwhile please let me know if you had any alternate approach that I can incorporate? Thanks!

Reply:
Sorry, I wasn't complaining about your use of `application.navigateToUrl` - I think its the right call - I suspect it should replace `history.push` as well. Both functions aim to do the same thing and we might as well use the same one in both places. more details - `packages/core/application/core-application-browser/src/contracts.ts:98`

Reply:
Hey @mattkime I updated to use `history.push` with `navigateToUrl` and updated tests too. Could you take look again please? Thanks! commit: https://github.com/elastic/kibana/pull/196787/commits/b71d3d2e6c2355be8ca2f495e38e4188d5707575